tlsconfig/tlsconfig_client.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 Palantir Technologies. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tlsconfig
+
+import (
+	"crypto/tls"
+	"fmt"
+)
+
+// TLSCertProvider is a function that returns a tls.Certificate used for TLS communication.
+type TLSCertProvider func() (tls.Certificate, error)
+
+// NewClientConfig returns a tls.Config that is suitable to use by a client in 2-way TLS connections configured with
+// the provided parameters. The provided TLSCertProvider is used as the source for the private key and certificate that
+// the client presents to servers.
+func NewClientConfig(tlsCertProvider TLSCertProvider, params ...ClientParam) (*tls.Config, error) {
+	if tlsCertProvider == nil {
+		return nil, fmt.Errorf("tlsCertProvider provided to NewClientConfig was nil")
+	}
+	configurers := []configurer{authKeyPairParam(tlsCertProvider)}
```

Comment:
It feels weird that this happens no matter what, but it may not make sense to skip this configuration.